src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -476,30 +476,38 @@ internal CompilationContext CreateCompilationContext(CSharpSyntaxNode syntax)
         private static void GetConstants(
             ArrayBuilder<LocalSymbol> builder,
             MethodSymbol method,
-            ImmutableArray<NamedLocalConstant> constants,
+            IEnumerable<ISymUnmanagedScope> scopes,
             MetadataDecoder metadataDecoder,
             ImmutableDictionary<string, ImmutableArray<bool>> dynamicLocalConstantMap,
             SourceAssemblySymbol containingAssembly)
         {
-            foreach (var constant in constants)
+            foreach (var scope in scopes)
             {
-                var info = metadataDecoder.GetLocalInfo(constant.Signature);
-                Debug.Assert(!info.IsByRef);
-                Debug.Assert(!info.IsPinned);
-                var type = info.Type;
-
-                ImmutableArray<bool> dynamicFlags;
-                if (dynamicLocalConstantMap != null && dynamicLocalConstantMap.TryGetValue(constant.Name, out dynamicFlags))
+                foreach (var constant in scope.GetConstants())
                 {
-                    type = DynamicTypeDecoder.TransformTypeWithoutCustomModifierFlags(
-                        type,
-                        containingAssembly,
-                        RefKind.None,
-                        dynamicFlags);
-                }
+                    string name = constant.GetName();
+                    object rawValue = constant.GetValue();
+                    var signature = constant.GetSignature();
 
-                var constantValue = ReinterpretConstantValue(constant.Value, type.SpecialType);
-                builder.Add(new EELocalConstantSymbol(method, constant.Name, type, constantValue));
+                    var info = metadataDecoder.GetLocalInfo(signature);
+                    Debug.Assert(!info.IsByRef);
+                    Debug.Assert(!info.IsPinned);
+                    var type = info.Type;
+
+                    var constantValue = PdbHelpers.GetConstantValue(type.EnumUnderlyingType(), rawValue);
```

Comment:
I see that it's an extension that does something different than the instance method... :-S

Please consider renaming to "EnumUnderlyingTypeOrSelf" (as in VB)